frontend/src/pages/pipelines/global/modelCustomization/SshKeyFileUpload.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  DropEvent,
+  FileUpload,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
+import React from 'react';
+import { ZodIssue } from 'zod';
+import { ZodErrorHelperText } from '~/components/ZodErrorFormHelperText';
+
+const MAX_SIZE = 1024 * 1024; // 1 MB as bytes
+
+export const SshKeyFileUpload: React.FC<{
+  onChange: (value: string) => void;
+  validationIssues: ZodIssue[];
+}> = ({ onChange, validationIssues }) => {
+  const [value, setValue] = React.useState('');
+  const [filename, setFilename] = React.useState('');
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [rejectedReason, setRejectedReason] = React.useState<string | undefined>();
+
+  const handleFileInputChange = (_event: DropEvent, file: File) => {
+    setFilename(file.name);
+    if (file.size > MAX_SIZE) {
+      setRejectedReason(`File size (${(file.size / 1024).toFixed(1)} KB) exceeds 1 MB limit`);
```

Comment:
I believe it would be more helpful to show the rejected file size in MB to make it easier to compare to the 1 MB limit.